lasagne/layers/conv.py

```diff
@@ -249,7 +249,7 @@ def get_output_for(self, input, input_shape=None, **kwargs):
                                       filter_shape=self.get_W_shape(),
                                       border_mode='full')
             shift = (self.filter_size[0] - 1) // 2
-            conved = conved[:, :, shift:input.shape[2] + shift]
+            conved = conved[:, :, shift:-shift]
```

Comment:
> but it seems like at some point that changed.

It's still the same, but it's only in `Conv2D??*Layer`, not in the original `Conv2DLayer`. To support it in `Conv2DDNNLayer` et al., we'd need to pad a little more than needed and crop the result, but right now we don't even remember `pad="same"` outside of the constructor. The easiest fix would be raising a `NotImplementedError` for even filter sizes.